src/components/timer.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import moment from 'moment-timezone';
+import DEFAULT_TEXT from '../constants/defaults';
+
+class Timer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      time: new Date(),
+    }
+    this.getCurrentTime = this.getCurrentTime.bind(this);
+    this.checkTimeDifference = this.checkTimeDifference.bind(this);
+    this.timerRef = React.createRef();
+  }
+
+  componentDidMount() {
+    this.timerRef.current.click();
+    this.initialTime = this.state.time;
+    this.timerInterval = setInterval(this.getCurrentTime, 1000);
+  }
+
+  componentDidUpdate(prevProps) {
+    if(prevProps.timezone !== this.props.timezone) {
+      this.clearTimer(this.timerInterval);
+      this.initialTime = new Date();
+      this.timerInterval = setInterval(this.getCurrentTime, 1000);
+    }
+  }
+
+  componentWillUnmount() {
+    this.clearTimer(this.timerInterval);
+  }
+
+  clearTimer(interval) {
+    clearInterval(interval);
+  }
+
+  emitSound = () => {
+    var utterMessage = new SpeechSynthesisUtterance(DEFAULT_TEXT.soundMessage);
+    window.speechSynthesis.speak(utterMessage);
+  }
+
+  checkTimeDifference() {
+    let currentTime = new Date();
```

Comment:
Why creating a new date ? Use state stored time ? Ure calling this post time update so it should have updated time